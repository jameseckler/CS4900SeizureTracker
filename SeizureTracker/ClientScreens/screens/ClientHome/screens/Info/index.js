import React, { Component } from 'react';
import { StyleSheet, Text, View, ImageBackground, Picker, TouchableOpacity, KeyboardAvoidingView, ScrollView, Image} from 'react-native';
import { ButtonGroup } from 'react-native-elements';
import 'firebase/firestore';
import {w, h, totalSize} from "../../../../../FirebaseLogin/api/Dimensions";
import SubmitButton from '../../screens/CreateLog/screens/LogType/components/SubmitButton';
import { Header } from 'react-navigation';

const background = require('../../../../../assets/background.png');


export default class Info extends Component{

    constructor () {

        super()
        this.state = {
            epilepsyInfo: true,
            seizureInfo: false,
            treatmentInfo: false,
            selectedIndex: 0
        }
        this.updateIndex = this.updateIndex.bind(this)
      }

    updateIndex (selectedIndex) {
      this.setState({selectedIndex})

      if(selectedIndex === 0){
        this.setState({epilepsyInfo: true});
        this.setState({seizureInfo: false});
        this.setState({treatmentInfo: false});
      }else if(selectedIndex === 1){
        this.setState({epilepsyInfo: false});
        this.setState({seizureInfo: true});
        this.setState({treatmentInfo: false});
      }else if(selectedIndex === 2){
        this.setState({epilepsyInfo: false});
        this.setState({seizureInfo: false});
        this.setState({treatmentInfo: true});
      }

    }
  
    render() {
      const buttons = ['Epilepsy', 'Seizures'];
      const { selectedIndex } = this.state;
      return(
          <ImageBackground source={background} style={{width: '100%', height: '100%'}}> 
              <View style={{ flex: 1, alignItems: 'center', flexDirection: 'column', 
              justifyContent: 'flex-start', marginTop: h(2)}}>
              <ButtonGroup
                onPress={this.updateIndex}
                selectedIndex={selectedIndex}
                buttons={buttons}
                containerStyle={{height: 20}}
              />
              {   
                  this.state.epilepsyInfo ?  
                  <KeyboardAvoidingView 
                      keyboardVerticalOffset = {Header.HEIGHT + 20} 
                      style={{flex: 1}}
                      behavior="padding">
                      <ScrollView>
                          <View style={styles.container}>
                            <Text style={styles.questionText}>
                              What are seizures?
                            </Text>
                            <Text style={styles.descriptionText}>
                              Seizures are one of the most frequently 
                              reported neurological conditions in dogs. 
                              A seizure may also be called a convulsion or fit, 
                              and is a temporary involuntary disturbance of 
                              normal brain function that is usually accompanied by 
                              uncontrollable muscle activity.
                            </Text>
                            <Text style={styles.questionText}>
                              What is Epilepsy?
                            </Text>
                            <Text style={styles.descriptionText}>
                              Epilepsy is a term used to describe 
                              repeated episodes of seizures. 
                              With epilepsy, the seizures can be single 
                              or may occur in clusters, 
                              and they can be infrequent and unpredictable 
                              or may occur at regular intervals.
                            </Text>
                          </View>
                      </ScrollView>
                  </KeyboardAvoidingView>
                  : null
              }
              {
                  this.state.seizureInfo ?  
                  <KeyboardAvoidingView 
                      keyboardVerticalOffset = {Header.HEIGHT + 20} 
                      style={{flex: 1}}
                      behavior="padding">
                      <ScrollView>
                          <View style={styles.container}>
                              <Text style={styles.questionText}>
                                What do I do if my pet has a seizure?
                              </Text>
                              <Text style={styles.descriptionText}>
                                Firstly, remain calm. Cats and dogs do not swallow their tongues, so you should not try to hold their tongue during the seizure. If the seizure 
                                occurs near a dangerous area such as stairs or a ledge, gently move them from the area.  
                                Know that your pet is not in pain or even conscious during the seizure, and that they may urinate or deficate.
                                </Text>
                                <Text style={styles.descriptionText}>
                                Try to commit to memory details such as the time of the seizure and how long it lasts. When creating logs in this application, you will 
                                be given a series of questions that your veterinarian can use to help create a treatment plan for your pet, so knowing the answers to these questions is helpful. 
                                You should create the seizure log as soon as you can, as details will be harder to remember the longer you wait.
                              </Text>
                              <Text style={styles.questionText}>
                                What should I do after the seizure?
                              </Text>
                              <Text style={styles.descriptionText}>
                                Contact your veterinarian as soon as possible. Seizures longer than 2-3 minutes may lead to hyperthermia, be ready with cold water or a wet towel 
                                and apply around your pet's body to help them cool down. If the seizure is longer than five minutes or if a second seizure occurs before complete 
                                consciousness, get emergency help for your pet immediately.
                              </Text>
                          </View>
                      </ScrollView>
                  </KeyboardAvoidingView>
                  : null
              }
              </View>
          </ImageBackground>
        );
    }
  }

  const styles = StyleSheet.create({
    linkVet: {
        color:'#ffffffEE',
        fontSize: totalSize(2),
        fontWeight: '700',
    },
        touchable: {
        alignSelf: 'flex-start',
        marginLeft: w(8),
        marginTop: h(1),
    },
    container: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      marginLeft: w(0),
    },
    create: {
      color:'white',
      fontSize: totalSize(2.4),
      marginTop: h(7),
      marginBottom: h(4),
      fontWeight: '700',
    },
    touchable: {
      alignSelf: 'flex-start',
      marginLeft: w(8),
      marginTop: h(1),
    },
    input: {
      marginVertical: h(1),
    },
    inputL: {
        marginVertical: h(1),
        marginRight: w(2),
    },
    inputR: {
        marginVertical: h(1),
        marginLeft: w(2),
    },
    check: {
      marginVertical: h(1),
      backgroundColor: 'transparent',
    },
    checkText: {
      color: 'white',
    },
    itemStyle: {
        fontSize: 15,
        height: 75,
        color: 'black',
        textAlign: 'center',
        fontWeight: 'bold'
      },
    picker: {
        width: 100
      },
    icon2: {
        width: w(94),
        height: h(5),
        marginTop: h(0), 
        marginBottom: h(0),
        flex: 1,
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'flex-start',
    },
    questionText: {
      color:'white',
      fontSize: totalSize(2.5),
      marginTop: h(7),
      marginBottom: h(4),
      fontWeight: '700',
    },
    descriptionText: {
      color:'white',
      fontSize: totalSize(1.5),
    },
  });