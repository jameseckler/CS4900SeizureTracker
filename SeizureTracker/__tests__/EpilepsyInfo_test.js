import React from 'react';
import renderer from 'react-test-renderer';
import Info from '../ClientScreens/screens/ClientHome/screens/Info';

test ("Create Log Button renders properly", () => {
    
    const rendered = renderer.create(<Info />).toJSON();
    expect(rendered).toBeTruthy();

});

it ("Test Create Log Button against snapshot", () => {

    const tree = renderer.create(<Info />).toJSON();
    expect(tree).toMatchSnapshot();

});