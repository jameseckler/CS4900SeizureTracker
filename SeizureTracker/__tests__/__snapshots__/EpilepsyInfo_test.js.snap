// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Create Log Button against snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginTop": 26.68,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#fff",
          "borderColor": "#e3e3e3",
          "borderRadius": 3,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 20,
          "marginBottom": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 5,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "borderBottomLeftRadius": 3,
            "borderRightColor": "#bdc6cf",
            "borderRightWidth": 0,
            "borderTopLeftRadius": 3,
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="buttonGroupItem"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#2089dc",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                }
              }
              testID="buttonGroupItemText"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Epilepsy
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "borderBottomRightRadius": 3,
            "borderLeftColor": "#bdc6cf",
            "borderLeftWidth": 1,
            "borderTopRightRadius": 3,
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "opacity": 1,
            }
          }
          testID="buttonGroupItem"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#5e6977",
                  "fontSize": 16.25,
                  "fontWeight": "500",
                }
              }
              testID="buttonGroupItemText"
              theme={
                Object {
                  "colors": Object {
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "primary": "#2196f3",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "primary": "#007aff",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#8F0CE8",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                }
              }
            >
              Seizures
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    />
  </View>
</View>
`;
